adapters/YahooAdapter/YahooAdapter/GADMAdapterYahooBaseClass.m

```diff
@@ -0,0 +1,335 @@
+}
+
+- (BOOL)isBannerAnimationOK:(GADMBannerAnimationType)animType {
+  return YES;
+}
+
+- (void)inlineAdDidLoad:(nonnull YASInlineAdView *)inlineAd {
+  NSLog(@"[YahooAdapter] Yahoo Mobile SDK loaded a banner ad successfully.");
+  dispatch_async(dispatch_get_main_queue(), ^{
+    self.inlineAd = inlineAd;
+    self.inlineAd.frame = CGRectMake(0, 0, inlineAd.adSize.width, inlineAd.adSize.height);
+    [self->_connector adapter:self didReceiveAdView:self.inlineAd];
+  });
+}
+
+- (void)inlineAdLoadDidFail:(nonnull YASInlineAdView *)inlineAd
+                  withError:(nonnull YASErrorInfo *)errorInfo {
+  NSLog(@"[YahooAdapter] Yahoo Mobile SDK failed to load a banner ad with error: %@",
+        errorInfo.description);
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [self->_connector adapter:self didFailAd:errorInfo];
+  });
+}
+
+- (void)inlineAdDidFail:(nonnull YASInlineAdView *)inlineAd
+              withError:(nonnull YASErrorInfo *)errorInfo {
+  NSLog(@"[YahooAdapter] Yahoo Mobile SDK returned an error for banner ad: %@",
+        errorInfo.description);
+  // This callback is forwarded when an error occurs in the ad's lifecycle.
+}
+
+- (void)inlineAdDidExpand:(nonnull YASInlineAdView *)inlineAd {
+  NSLog(@"[YahooAdapter] Yahoo Mobile SDK expanded a banner ad.");
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [self->_connector adapterWillPresentFullScreenModal:self];
+  });
+}
+
+- (void)inlineAdDidCollapse:(nonnull YASInlineAdView *)inlineAd {
+  NSLog(@"[YahooAdapter] Yahoo Mobile SDK collapsed a banner ad.");
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [self->_connector adapterDidDismissFullScreenModal:self];
+  });
+}
+
+- (void)inlineAdClicked:(nonnull YASInlineAdView *)inlineAd {
+  NSLog(@"[YahooAdapter] Yahoo Mobile SDK recorded a click on a banner ad.");
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [self->_connector adapterDidGetAdClick:self];
+  });
+}
+
+- (void)inlineAdDidLeaveApplication:(nonnull YASInlineAdView *)inlineAd {
+  NSLog(@"[YahooAdapter] Yahoo Mobile SDK has caused the user to leave the application from a "
+        @"banner ad.");
+}
+
+- (void)inlineAdDidResize:(nonnull YASInlineAdView *)inlineAd {
+  // AdMob does not expose a resize callback to map to this.
+}
+
+- (void)inlineAdDidRefresh:(nonnull YASInlineAdView *)inlineAd {
+  // AdMob publishers use the AdMob inline refresh mechanism, so an implementation here is not
+  // needed.
+}
+
+- (nullable UIViewController *)inlineAdPresentingViewController {
+  return [_connector viewControllerForPresentingModalView];
+}
+
+- (void)inlineAd:(nonnull YASInlineAdView *)inlineAd
+           event:(nonnull NSString *)eventId
+          source:(nonnull NSString *)source
+       arguments:(nonnull NSDictionary<NSString *, id> *)arguments {
+  // A generic callback that does currently need an implementation for inline placements.
+}
+
+#pragma mark - Interstital
+
+- (void)getInterstitial {
```

Comment:
I verified interstitial with one of our test ads, the interstitial displays and all callbacks are working correctly.